api/v1/monitor_types.go

```diff
@@ -17,11 +17,78 @@ limitations under the License.
 package v1
 
 import (
+	v1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // MonitorSpec defines the desired state of Tigera monitor.
 type MonitorSpec struct {
+	// ExternalPrometheus optionally configures integration with an external Prometheus for scraping Calico metrics. When
+	// specified, the operator will render resources in the defined namespace. This option can be useful for configuring
+	// scraping from git-ops tools without the need of post-installation steps.
+	ExternalPrometheus *ExternalPrometheus `json:"externalPrometheus,omitempty"`
+}
+
+type ExternalPrometheus struct {
+	// ServiceMonitor when specified, the operator will create a ServiceMonitor object in the namespace. It is recommended
+	// that you configure labels if you want your prometheus instance to pick up the configuration automatically.
+	// The operator will configure 1 endpoint by default:
+	// - Params to scrape all metrics available in Calico Enterprise.
+	// - BearerTokenSecret (If not overridden, the operator will also create corresponding RBAC that allows authz to the metrics.)
+	// - TLSConfig, containing the caFile and serverName.
+	// +optional
+	ServiceMonitor *ServiceMonitor `json:"serviceMonitor,omitempty"`
+
+	// Namespace is the namespace where the operator will create resources for your Prometheus instance. As long as the
+	// namespace does not exist, nothing will be rendered.
```

Comment:
I think this wording should be "The namespace must be created in advance before the Prometheus resources will be created" - "rendered" isn't a term that will resonate with end users, I think.